modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/cluster/services/Cluster.scala

```diff
@@ -85,16 +85,18 @@ object Cluster {
 
       def info(implicit hasher: Hasher[F]): F[Set[PeerInfo]] =
         getRegistrationRequest.flatMap { req =>
+
           def self = PeerInfo(
             req.id,
             req.ip,
             req.publicPort,
             req.p2pPort,
             req.session.value.toString,
+            req.clusterSession.value.toString,
             req.state
           )
 
-          clusterStorage.getResponsivePeers.map(_.map(PeerInfo.fromPeer) + self)
+          clusterStorage.getResponsivePeers.map(_.map(PeerInfo.fromPeer(req.clusterSession)) + self)
```

Comment:
You put node (self) cluster session as some other peer's cluster session. Instead, you should add clusterSession to the `Peer` case class.

Reply:
the initial idea was use just the self session, but your suggestion makes sense

Reply:
IMO it does not make sense if you'd use the self session to present as some other node's session

Reply:
Shouldn't be all in the same session? Anyway, that code is outdated, and it takes the session from whatever is in the Peer object

Reply:
Well, they "should" :) The only check is made during Joining process. If we would change that in the future, your code would support it now, instead of blind assumptions :)

Reply:
yeah, the code now reports the cluster session id that the peer had at the time of joining.